x-pack/legacy/plugins/integrations_manager/server/config.ts

```diff
@@ -0,0 +1,48 @@
+/*
+ * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+ * or more contributor license agreements. Licensed under the Elastic License;
+ * you may not use this file except in compliance with the Elastic License.
+ */
+
+import { schema, TypeOf } from '@kbn/config-schema';
+
+export const config = {
+  schema: schema.object({
+    enabled: schema.boolean(),
+    registryUrl: schema.string(),
+  }),
+};
+
+export type IntegrationsManagerConfigSchema = TypeOf<typeof config.schema>;
+
+const DEFAULT_CONFIG = {
+  enabled: true,
+  registryUrl: 'http://integrations-registry.app.elstc.co',
+};
+
+// As of 2019, this is a Singleton because of the way JavaScript modules are specified.
+// Every other module that imports this file will have access to the same object.
+
+// This is meant to be only updated from the config$ Observable's subscription
+// (see the Plugin class constructor in server/plugin.ts) but this is not enforced.
+
+class IntegrationsManagerConfigStore {
```

Comment:
Another option is not use a `class` and export an object which only has the methods we want exposed.

```typescript
let config: IntegrationsManagerConfigSchema = DEFAULT_CONFIG;

export const integrationsManagerConfigStore = {
  updateConfig(newConfig: IntegrationsManagerConfigSchema) {
    config = Object.assign({}, config, newConfig);
  },
  getConfig() {
    return config;
  }
}
```

While `private` does prevent TS consumers from accessing `.config` directly, it's still available in the JS.

I'm fine with either but this less code, possibly more secure, and doesn't seem like we need a class.